### PR TITLE
Ideal Indirection SP2020

### PR DESCRIPTION
**Pull request by HungWoeiNeoh**
Change notes:
- Revised docs to reflect how the assignment is currently
- Rephrased some parts of the documentation
- Enabled assignment to be visible in the course website

### CODE REVIEW

File: _docs/ideal_indirection.md [hunk 1]
```diff
@@ -14,32 +14,52 @@ wikibook:
...
 
-In this assignment, you will be working on managing a mapping of 32-bit virtual address space  to a 32-bit physical address space with 4 KB (kilobyte) pages.
+In this assignment, you will be working on managing a mapping of a 32-bit virtual address space to a 32-bit physical address space with 4 KB (kilobyte) pages.
 Each component of virtual memory has been split up into several files to help you build a mental model of virtual memory.
-Reading through these files can start to help you understand the roles of the different hardware and software involved in managing virtual memory.
-You will only have to write two functions in `mmu.c`, but it requires a good understanding of the coursebook and decent knowledge of the rest of the provided code.
+Reading through these files will help you understand the roles of the different hardware and software involved in managing virtual memory.
+You will only have to write two functions in `mmu.c`, but it requires a good understanding of the virtual memory translation process and decent understanding of the rest of the provided code.
+
+## How To Tackle This Assignment
```
Comment (PradyumnaShome) [inline _docs/ideal_indirection.md]: +1 - thanks for adding :)

File: _docs/ideal_indirection.md [hunk 2]
```diff
@@ -14,32 +14,52 @@ wikibook:
...
+  * raising page faults and segfaults
+  * requesting memory frames from the kernel and retrieving swap files from the disk
+  * other stuff
+* Once this is done, turn these steps into actual code. Then, test with the given test cases.
 
 The rest of this documentation will serve to help you understand the purpose of each file and give you a high level understanding of virtual memory.
-The implementation details of each function are documented in the header and source files.
-It is your responsibility to read every line of code before starting this assignment, since you will not be able to do anything meaningful otherwise.
+The implementation details of each function are documented in the header files.
+The recommended sequence of reading the provided files will follow the sequence of files introduced in this documentation.
```
Comment (PradyumnaShome) [inline _docs/ideal_indirection.md]: I hope students follow this...for their own good...

File: _docs/ideal_indirection.md (hunk 1 above)
Comment (HungWoeiNeoh) [reply _docs/ideal_indirection.md]: You know an assignment is special when that is added

File: _docs/ideal_indirection.md (hunk 2 above)
Comment (HungWoeiNeoh) [reply _docs/ideal_indirection.md]: I hope so too. Anyone who jumps straight into `mmu.h` without reading the other files will have a headache

**Pull request merged**